pkg/awsutils/awsutils.go

```diff
@@ -1399,13 +1383,13 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddress(eniID string) error {
 }
 
 func (cache *EC2InstanceMetadataCache) FetchInstanceTypeLimits() error {
-	_, ok := InstanceNetworkingLimits[cache.instanceType]
+	_, ok := vpc.GetInstance(cache.instanceType)
 	if ok {
 		return nil
 	}
 
 	log.Debugf("Instance type limits are missing from vpc_ip_limits.go hence making an EC2 call to fetch the limits")
-	var eniLimits InstanceTypeLimits
+	var eniLimits vpc.InstanceTypeLimits
```

Comment:
I don't have a great suggestion here, but it feels to me like we shouldn't be creating a `vpc.InstanceTypeLimits` object and passing to `vpc.SetInstance`. It seems like we should be passing the relevant values and the `vpc` package should create and add the struct. Then it is more following a constructor package. Maybe `vpc` package needs a `New` function that takes all of the fields as arguments